cmp/cmpopts/transform.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2017, The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE.md file.
+
+package cmpopts
+
+import (
+	"reflect"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/internal/value"
+)
+
+// DiscardMapZeros returns a Transformer option that discards all map entries
+// with a zero value. This is useful for comparing two maps in such a way
+// that entries with the zero value are ignored.
+//
+// DiscardMapZeros can be used in conjunction with EquateEmpty,
+// but cannot be used with SortMaps.
+func DiscardMapZeros() cmp.Option {
```

Comment:
DiscardZeroElements also operating on slices feels like it is over-stretching what it should do. Although, I admit that it probably would not break anything since zero-values of any type is typically equal to itself. I do like the fact that I wouldn't need to think about whether this applies to maps or slices.

Let me think about this.